### ISSUE TITLE
[Bug]: ShaftEngine_ElementActions_shaft waited long time before clicking an existing element. 

### ISSUE THREAD

**Issue by zanoon2020** [ShaftHQ/SHAFT_ENGINE#910]
**Describe the bug**

before clicking the element shaft waited the defined duration in timeout property file > defaultElementIdentificationTimeout , 
 
the element :   
![image](https://user-images.githubusercontent.com/65794900/221358241-26c87da9-4477-4b1f-a84d-d969a5e3860b.png)
 
the locator that cause the problem :
 By.xpath("//app-checkbox[@name='prepaid']//input") 

another locator for same element but didn't cause the issue 
 By.xpath("//app-checkbox[@name='prepaid']//mat-checkbox")

**Link to SHAFT_Engine Console logs**
https://gist.github.com/zanoon2020/ed8cfbfb312582988eabde07b6795699
 
time stamps for the issue : 
[INFO ] 2023-02-25 14:47:18.211
[INFO ] 2023-02-25 14:48:19.552

**Environment**

 1. SHAFT_Engine version that exhibits the issue (Note: We support only the latest release): 7.1.20230222
 2. Last SHAFT_Engine version that did not exhibit the issue (if applicable):
 3. Desktop OS: windows 10
 4. Desktop Browser (if applicable): [e.g. chrome, safari] chrome
 5. Desktop Browser Version (if applicable): [e.g. 22] 110.0.5481.178 

**To Reproduce**

**Expected behavior**

**Screenshots**

If applicable, add screenshots, gifs, or video recordings to help explain the issue you're facing.

**Additional context**

Add any other relevant context about the problem here.

**Comment by MohabMohie** [comment]
first of all "THANK YOU" for the detailed description and the debug logs! you even enabled debugging and sent me a timestamp! I really appreciate that :D that's the best bug report I've even received.

now, investigating the debug logs, I see that SHAFT kept trying to:
findElement > which was succeeding
and then clickElement > which was consistently failing like the following example
```
[DEBUG] 2023-02-25 14:48:18.480 [AsyncHttpClient-1-2] ChannelManager - Adding key: http://localhost:23406 for channel [id: 0x64ed172b, L:/127.0.0.1:51246 - R:localhost/127.0.0.1:23406]
[DEBUG] 2023-02-25 14:48:18.481 [Forwarding clickElement on session 691d718f31216c12474b42a0b35fc261 to remote] NettyRequestSender - Using pooled Channel '[id: 0x64ed172b, L:/127.0.0.1:51246 - R:localhost/127.0.0.1:23406]' for 'POST' to 'http://localhost:23406/session/691d718f31216c12474b42a0b35fc261/element/11bafa21-d932-4fcc-b4fd-a510336fd267/click'
[DEBUG] 2023-02-25 14:48:18.481 [Forwarding clickElement on session 691d718f31216c12474b42a0b35fc261 to remote] NettyRequestSender - Using open Channel [id: 0x64ed172b, L:/127.0.0.1:51246 - R:localhost/127.0.0.1:23406] for POST '/session/691d718f31216c12474b42a0b35fc261/element/11bafa21-d932-4fcc-b4fd-a510336fd267/click'
[DEBUG] 2023-02-25 14:48:19.509 [AsyncHttpClient-1-2] HttpHandler - 

Request DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)
POST /session/691d718f31216c12474b42a0b35fc261/element/11bafa21-d932-4fcc-b4fd-a510336fd267/click HTTP/1.1
User-Agent: selenium/4.8.1 (java windows)
Content-Length: 50
Content-Type: application/json; charset=utf-8
host: localhost:23406
accept: */*

Response DefaultHttpResponse(decodeResult: success, version: HTTP/1.1)
HTTP/1.1 400 Bad Request
Content-Length: 1739
Content-Type: application/json; charset=utf-8
cache-control: no-cache

```
finally after the timeout expires, the engine tries to use JavaScript and succeeds, so I'm assuming that you have "forceClickUsingJavascript" enabled
```
[DEBUG] 2023-02-25 14:48:19.520 [AsyncHttpClient-1-2] ChannelManager - Adding key: http://localhost:23406 for channel [id: 0x64ed172b, L:/127.0.0.1:51246 - R:localhost/127.0.0.1:23406]
[DEBUG] 2023-02-25 14:48:19.520 [Forwarding executeScript on session 691d718f31216c12474b42a0b35fc261 to remote] NettyRequestSender - Using pooled Channel '[id: 0x64ed172b, L:/127.0.0.1:51246 - R:localhost/127.0.0.1:23406]' for 'POST' to 'http://localhost:23406/session/691d718f31216c12474b42a0b35fc261/execute/sync'
[DEBUG] 2023-02-25 14:48:19.520 [Forwarding executeScript on session 691d718f31216c12474b42a0b35fc261 to remote] NettyRequestSender - Using open Channel [id: 0x64ed172b, L:/127.0.0.1:51246 - R:localhost/127.0.0.1:23406] for POST '/session/691d718f31216c12474b42a0b35fc261/execute/sync'
[DEBUG] 2023-02-25 14:48:19.551 [AsyncHttpClient-1-2] HttpHandler - 

Request DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)
POST /session/691d718f31216c12474b42a0b35fc261/execute/sync HTTP/1.1
User-Agent: selenium/4.8.1 (java windows)
Content-Length: 228
Content-Type: application/json; charset=utf-8
host: localhost:23406
accept: */*

Response DefaultHttpResponse(decodeResult: success, version: HTTP/1.1)
HTTP/1.1 200 OK
Content-Length: 14
Content-Type: application/json; charset=utf-8
cache-control: no-cache
```

From the tree you shared, I think the issue is as follows:
1. SHAFT is designed to wait until an element is interactable, and this element is not interactable, it's throwing an invalidelementstate exception, which is why SHAFT keeps looping until the timeout expires.
2. If you can, let's disable "clickUsingJavaScript" and try again, SHAFT should fail and provide a staktrace that should show the invalidelementstate as the root cause.
3. This is usually caused by FE developers not propagating actions properly... the action (method) is actually placed on the wrapper element (the locator that's working for you) and not on the input element, because a regular user wouldn't be able to click an element that they cannot see, so this issue is never reported... but it impacts automation.
4. The correct solution is to adjust your locator, and always use the upper most element locator. HOWEVER, I think it makes sense to add a public "clickUsingJavascript()" method in SHAFT's element actions, so that users can use it whenever they need to.

In conclusion, I think this is working as designed, so I'll be closing this bug. However, I will refer to this bug and open a new feature request to expose clickUsingJavaScript() in the fluent element actions class. This will be an easy thing to do since the function is already in the code, it's just not exposed.
